### PR TITLE
feat(ci): reduce complexity of DCO check

### PR DESCRIPTION
**Pull request by Neudrino**
## Summary

- This is a follow up to https://github.com/magma/magma/pull/14474 with in principle the same simplifications.

## Test plan

- See message in this PR
- CI: https://github.com/magma/magma/actions/workflows/dco-check.yml?query=branch%3Aci%2Frefactor-pr-messages
  - :heavy_check_mark: Check is :green_circle: https://github.com/magma/magma/actions/runs/3487912178
  - :heavy_check_mark: Check is :red_circle: https://github.com/magma/magma/actions/runs/3487937403

### CODE REVIEW

File: .github/workflows/dco-check.yml [hunk 1]
```diff
@@ -11,69 +11,36 @@
 
 name: PR Check DCO
 on:
-  pull_request:
-    types: [ opened, reopened, synchronize ]
```
Comment (Neudrino) [inline .github/workflows/dco-check.yml:15]: This is actually [the default](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) and can be omitted.

**Review by LKreutzer — COMMENTED** [review]
I tested these changes by pushing the commit to the master branch of my fork and open some PR to that branch.

Whether the PR commit is signed or not the workflow run is a failure. The reason is that `tim-actions/get-pr-commits` only supports pull request events and not `pull_request_target` (see [source code](https://github.com/tim-actions/get-pr-commits/blob/dd6aa17f3353ab68cfc827ce36d8d7f2f02e831a/index.js)).
Logging from failed runs:
Screenshot from 2023-01-04 11-46-59

Link to runs: 
- https://github.com/LKreutzer/magma/actions/runs/3837218178/jobs/6532271147
- https://github.com/LKreutzer/magma/actions/runs/3837228957/jobs/6532291495

Link to test PR: https://github.com/LKreutzer/magma/pull/12

Options:
- Is it possible to use a `pull_request` trigger for this workflow? (I will test this)
- Use a different action that supports this trigger.
- Close this PR if none of these are feasible.

**Comment by Neudrino** [comment]
Thanks for your effort @LKreutzer !
If it is too much effort, feel free to close the PR.

I originally thought this would be a easy and quick simplification of the current workflow, however, the devil is in the details. :see_no_evil:

**Comment by LKreutzer** [comment]
Update: It seems that a `pull_request` trigger works.
Green: https://github.com/LKreutzer/magma/actions/runs/3837349983/jobs/6532554231
Red: https://github.com/LKreutzer/magma/actions/runs/3837360198/jobs/6532578181

Is there any reason not to use `pull_request`?

**Comment by Neudrino** [comment]
> Is there any reason not to use `pull_request`?

The `pull_request` trigger only works for the **non-forked** PRs (as e.g. this one). For forked ones you would need the `pull_request_target` due to the credential access with write permission on the PR for the comments.

**Comment by LKreutzer** [comment]
> > Is there any reason not to use `pull_request`?
> 
> The `pull_request` trigger only works for the **non-forked** PRs (as e.g. this one). For forked ones you would need the `pull_request_target` due to the credential access with write permission on the PR for the comments.

Got it and thank you for the quick response! In that case I would suggest to close this PR as another solution would be needed to make this work.

**Comment by Neudrino** [comment]
In the bigger picture, one probably still needs the split in two workflows with the `workflow_run` trigger.
I originally hoped to get rid of it. Due to the insync check this apparently is not possible anyway.

**Pull request closed without merging**